lang/syn/src/codegen/accounts/constraints.rs

```diff
@@ -715,15 +841,74 @@ fn generate_constraint_init_group(
+                                    let cpi_program = #token_program.to_account_info();
+                                    let accounts = ::anchor_spl::token_interface::MetadataPointerInitialize {
+                                        token_program_id: #token_program.to_account_info(),
+                                        mint: #field.to_account_info(),
+                                    };
+                                    let cpi_ctx = anchor_lang::context::CpiContext::new(cpi_program, accounts);
+                                    ::anchor_spl::token_interface::metadata_pointer_initialize(cpi_ctx, #metadata_pointer_authority, #metadata_pointer_metadata_address)?;
+                                },
+                                ::anchor_spl::token_interface::spl_token_2022::extension::ExtensionType::MintCloseAuthority => {
+                                    let cpi_program = #token_program.to_account_info();
+                                    let accounts = ::anchor_spl::token_interface::MintCloseAuthorityInitialize {
+                                        token_program_id: #token_program.to_account_info(),
+                                        mint: #field.to_account_info(),
+                                    };
+                                    let cpi_ctx = anchor_lang::context::CpiContext::new(cpi_program, accounts);
+                                    ::anchor_spl::token_interface::mint_close_authority_initialize(cpi_ctx, #close_authority)?;
+                                },
+                                ::anchor_spl::token_interface::spl_token_2022::extension::ExtensionType::TransferHook => {
+                                    let cpi_program = #token_program.to_account_info();
+                                    let accounts = ::anchor_spl::token_interface::TransferHookInitialize {
+                                        token_program_id: #token_program.to_account_info(),
+                                        mint: #field.to_account_info(),
+                                    };
+                                    let cpi_ctx = anchor_lang::context::CpiContext::new(cpi_program, accounts);
+                                    ::anchor_spl::token_interface::transfer_hook_initialize(cpi_ctx, #token_hook_authority, #token_hook_program_id)?;
+                                },
+                                _ => {} // do nothing
+                            }
+                        };
+                    }
+
+                    if !#if_needed || owner_program == &anchor_lang::solana_program::system_program::ID {
+                        // Define payer variable.
+                        #payer_optional_check
 
                         // Initialize the mint account.
-                        let cpi_program = #token_program.to_account_info();
+                        let cpi_program = #token_program.clone().to_account_info();
```

Comment:
```suggestion
                        let cpi_program = #token_program.to_account_info();
```

`to_account_info` already returns an owned `AccountInfo`:

https://github.com/coral-xyz/anchor/blob/08110e63fa5a7369830db972d4abdcfa1aaa7f2e/lang/src/lib.rs#L175-L177